be/src/olap/types.cpp

```diff
@@ -259,7 +259,7 @@ TypeInfoPtr get_type_info(const TabletColumn* col) {
         const auto* val_column = &col->get_sub_column(1);
         TypeInfoPtr value_type = get_type_info(val_column);
         MapTypeInfo* map_type_info = new MapTypeInfo(std::move(key_type), std::move(value_type));
-        return create_static_type_info_ptr(map_type_info);
+        return create_dynamic_type_info_ptr(map_type_info);
```

Comment:
consider ARRAY

Reply:
ARRAY did not has this problem , get_array_type_info(child_column->type(), iterations) which return a with  
```static const ArrayTypeInfo*``` which is static .